### DISCUSSION TITLE
Linkerd v2 HPA(metrics server) issue.

### DISCUSSION THREAD

**Discussion by mumirza44** [linkerd/linkerd2#11484] [Help]
Hi,
I am facing issues with HPA after injecting linkerd proxy in my application pods. The HPA targets were showing unknown status.
Then I set the **config.linkerd.io/proxy-cpu-request** annotation in my deployment and the HPA was working. but it seems the linkerd proxy is the default container. so my deployments are scaling based on linkerd proxy cpu usage. I would like to make my application container as the default container of the pod so that i can use HPA with it.
Kindly let me know what can be done in this situation and what is the best approach of using HPA with pods injected with linkerd proxy. I am using metrics server.

**Comment by wmorgan** [comment]
Can't you specify an alternate container for HPA?
